Testnet/Airdrop/Airdrop/Airdrop.cs

```diff
@@ -0,0 +1,202 @@
+    {
+        return PersistentState.GetStruct<Status>($"Status:{address}");
+    }
+
+    /// <summary>Sets the status for a given address.</summary>
+    private void SetAccountStatus(Address address, Status status)
+    {
+        PersistentState.SetStruct($"Status:{address}", status);
+    }
+
+    /// <summary>Validates and registers accounts. See <see cref="AddRegistrantExecute(Address)"/></summary>
+    public bool Register()
+    {
+        return AddRegistrantExecute(Message.Sender);
+    }
+
+    /// <summary>Allows owner of the contract to manually add a new registrant. See <see cref="AddRegistrantExecute(Address)"/></summary>
+    public bool AddRegistrant(Address registrant)
+    {
+        return Message.Sender == Owner && AddRegistrantExecute(registrant);
+    }
+
+    /// <summary>Allows owner to close registration for the airdrop at any time.</summary>
+    public bool CloseRegistration()
+    {
+        if (Message.Sender != Owner)
+        {
+            return false;
+        }
+
+        RegistrationIsClosed = true;
+
+        return true;
+    }
+
+    /// <summary>
+    /// Withdraw funds after registration period has closed. Validates account status and calls the tokens contract
+    /// address that is being airdropped to transfer amount to sender. On success, set senders new status and log it.
+    /// </summary>
+    public bool Withdraw()
+    {
+        bool invalidAccountStatus = GetAccountStatus(Message.Sender) != Status.ENROLLED;
+        if (invalidAccountStatus || !RegistrationIsClosed || AmountToDistribute == 0)
+        {
+            return false;
+        }
+
+        var transferParams = new object[] { Message.Sender, AmountToDistribute };
+
+        ITransferResult result = Call(TokenContractAddress, AmountToDistribute, "TransferTo", transferParams);
+
+        if (result == null || !result.Success)
+        {
+            return false;
+        }
+
+        SetAccountStatus(Message.Sender, Status.FUNDED);
+
+        Log(new StatusLog { Registrant = Message.Sender, Status = Status.FUNDED });
+
+        return true;
+    }
+
+    /// <summary>Validates and adds a new registrant. Updates the NumberOfRegistrants, account status, and logs result.</summary>
+    private bool AddRegistrantExecute(Address registrant)
+    {
+        if (registrant == Owner)
+        {
+            return false;
+        }
+
+        bool invalidAddressStatus = GetAccountStatus(registrant) != Status.NOT_ENROLLED;
+        if (invalidAddressStatus || RegistrationIsClosed || NumberOfRegistrants >= this.TotalSupply)
+        {
+            return false;
+        }
+
+        NumberOfRegistrants += 1;
+
+        SetAccountStatus(registrant, Status.ENROLLED);
+
+        Log(new StatusLog { Registrant = registrant, Status = Status.ENROLLED });
+
+        return true;
+    }
+
+    public struct StatusLog
+    {
+        [Index]
+        public Address Registrant;
+
+        public Status Status;
+    }
+
+    public enum Status
```

Comment:
Not necessary, but would recommend explicitly specifying the enum type here, `public enum Status : uint`